src/solver/simulation/common-hydro-levels.cpp

```diff
@@ -94,6 +94,15 @@ void updatingWeeklyFinalHydroLevel(const Data::AreaList& areas, PROBLEME_HEBDO&
         const std::vector<double>& niv = weeklyResults.niveauxHoraires;
 
         problem.previousSimulationFinalLevel[index] = niv[nbHoursInAWeek - 1];
+
+        // Tiny numerical errors (for instance when using solver Xpress) can lead
+        // to a final level slightly above reservoir capacity.
+        // This causes an infeasibility when solving weekly optimization associated to
+        // week next to the current one. We make sure that final level remains correct.
+        if (problem.previousSimulationFinalLevel[index] > area->hydro.reservoirCapacity)
```

Comment:
Do the same if the final level falls below 0

Reply:
done (see below), but we could clearly use C++17's [`std::clamp`](https://en.cppreference.com/w/cpp/algorithm/clamp.html) for a nice one-liner.

```cpp
problem.previousSimulationFinalLevel[index] = std::clamp(niv[nbHoursInAWeek - 1],
                                                         0.,
                                                         area->hydro.reservoirCapacity);

```

Reply:
We're on discussion with @tbittar and @Juliette-Gerbaux , this change is about to be reverted.
I turned the PR into draft mode.